messages-list-directive.js

```diff
@@ -19,42 +19,148 @@ function factory(brMessagesService, $location) {
   function Link(scope) {
     var model = scope.model = {};
     model.messages = null;
-
+    model.loading = true;
     model.sorting = {
-      subject: '+',
       date: '-'
     };
-    model.orderBy = ['-content.date', '+_id'];
+    model.orderBy = ['-date', '+id'];
+    model.showDeleteSuccessAlert = false;
+    model.showDeleteFailAlert = false;
+    if(brMessagesService.returningFrom === 'archive') {
+      model.inboxActive = false;
+      model.archiveActive = true;
+      brMessagesService.returningFrom = null;
+    } else {
+      model.inboxActive = true;
+      model.archiveActive = false;
+    }
+
+    var allMessages = [];
+    processMessageLists();
+
+    // Update the message service with the list's current URL so we can path back to it later
+    brMessagesService.setMessageListUrl($location.path());
 
     brMessagesService.getAll()
       .then(function(result) {
-        model.messages = result.data;
+        allMessages = result.data;
+        for(var message in allMessages) {
```

Comment:
Don't use `for x in y` with arrays. Use `forEach` or `for(var i = 0; i < allMessages.length; ++i)`